### PR TITLE
Moving image component into the mc-components repo

### PR DESCRIPTION
**Pull request by stephenp**
## Overview
We've been requested to integrate the `Image` component from the main repo into mc-components.  This brings in the existing component and updates the `TileImage` component to use this new component.

## Risks
Low

## Changes
New component

## Issue
N/A

## Breaking change?
Backwards compatible

### CODE REVIEW

File: src/components/Button/index.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ export default class Button extends PureComponent {
     className: PropTypes.string,
     fullWidth: PropTypes.bool,
     kind: PropTypes.oneOf([
-      'applepay',
+      'apple',
```
Comment (stephenp) [inline src/components/Button/index.js:24]: Snuck this in - the `applepay` type was updated, but missed here

File: src/components/Button/index.stories.js [hunk 2]
```diff
@@ -143,9 +143,9 @@ class ButtonStory extends PureComponent {
             <PropExample
               name='kind'
               type='string'
-              description={`
+              description="
```
Comment (stephenp) [inline src/components/Button/index.stories.js:146]: console warning fix

File: src/components/Button/index.stories.js [hunk 3]
```diff
@@ -335,13 +335,13 @@ class ButtonStory extends PureComponent {
                 </div>
 
                 <div className='col-6'>
-                  <Button secondary fullWidth>
+                  <Button kind='secondary' fullWidth>
```
Comment (stephenp) [inline src/components/Button/index.stories.js:338]: console warning fix

File: src/components/Carousel/index.stories.js [hunk 4]
```diff
@@ -77,7 +77,9 @@ storiesOf('Components|Carousel', module)
     <ResponsiveHandler>
       {({ responsive }) =>
         <div className='container'>
-          <DocHeader title='Carousel' />
+          <DocHeader
```
Comment (stephenp) [inline src/components/Carousel/index.stories.js:80]: console warning fix

File: src/components/Carousel/props.stories.js [hunk 5]
```diff
@@ -87,7 +87,9 @@ storiesOf('Components|Carousel', module)
     <ResponsiveHandler>
       {({ responsive }) =>
         <div className='container'>
-          <DocHeader title='Carousel' />
+          <DocHeader
```
Comment (stephenp) [inline src/components/Carousel/props.stories.js:90]: console warning fix

File: src/components/Image/index.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
+import React from 'react'
+import { number, string, oneOf, oneOfType } from 'prop-types'
+import cn from 'classnames'
+import { debounce } from 'lodash'
+import resizeImage from './resize-image'
+
+const ASPECT_RATIOS = ['1x1', '2x3', '3x4', '4x3', '9x16', '16x9', '21x9']
```
Comment (jstnjns) [inline src/components/Image/index.js:7]: https://github.com/yankaindustries/mc-components/blob/main/src/components/Tile/index.js#L6

**Review by jstnjns — APPROVED** [review]

**Pull request merged**